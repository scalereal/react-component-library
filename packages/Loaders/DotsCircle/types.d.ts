import { HTMLProps, Omit, WithThemeProp } from '../../utils';
export declare type AvatarSizes = 'XXS' | 'XS' | 'S' | 'M' | 'L' | 'XL';
export interface Props extends Omit<HTMLProps<HTMLDivElement>, 'size'>, WithThemeProp {
    display?: boolean;
    size?: AvatarSizes;
    Color?: string;
}
