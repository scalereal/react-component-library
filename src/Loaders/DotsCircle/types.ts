import { HTMLProps, Omit, WithThemeProp } from '../../utils';

export type AvatarSizes = 'XXS' | 'XS' | 'S' | 'M' | 'L' | 'XL' ;

export interface Props extends Omit<HTMLProps<HTMLDivElement>,'size'>,WithThemeProp {
    display?:boolean;
    size?: AvatarSizes;
    Color?:string;
}