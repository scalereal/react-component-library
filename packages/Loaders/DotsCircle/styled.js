var __makeTemplateObject = (this && this.__makeTemplateObject) || function (cooked, raw) {
    if (Object.defineProperty) { Object.defineProperty(cooked, "raw", { value: raw }); } else { cooked.raw = raw; }
    return cooked;
};
import styled from 'styled-components';
export var Loader = styled.div(templateObject_1 || (templateObject_1 = __makeTemplateObject(["\n    display: inline-block;\n    width: 1em;\n    height: 1em;\n    color: ", ";\n    vertical-align: middle;\n    pointer-events: none;\n    border-radius: 50%;\n    animation: 1s loader-01 linear infinite;\n\n    @keyframes loader-01 {\n        0% {\n            transform: rotate(0deg);\n        }\n        100% {\n            transform: rotate(360deg);\n        }\n    }\n    \n"], ["\n    display: inline-block;\n    width: 1em;\n    height: 1em;\n    color: ", ";\n    vertical-align: middle;\n    pointer-events: none;\n    border-radius: 50%;\n    animation: 1s loader-01 linear infinite;\n\n    @keyframes loader-01 {\n        0% {\n            transform: rotate(0deg);\n        }\n        100% {\n            transform: rotate(360deg);\n        }\n    }\n    \n"])), function (pr) { return pr.Color ? pr.Color : 'inherit'; });
var mixin = function (size) {
    switch (size) {
        case "XXS":
            return "font-size: 0.8rem;\n            " + Loader + "{\n                border: 0.188em dotted currentcolor;\n            }";
        case "XS":
            return "font-size: 1rem;\n            " + Loader + "{\n                border: 0.188em dotted currentcolor;\n            }\n            ";
        case "S":
            return "font-size: 2rem;\n            " + Loader + "{\n                border: 0.2em dotted currentcolor;\n            }";
        case "M":
            return "font-size: 3rem;\n            " + Loader + "{\n                border: 0.2em dotted currentcolor;\n            }";
        case "L":
            return "font-size: 5rem;\n            " + Loader + "{\n                border: 0.2em dotted currentcolor;\n            }";
        case "XL":
            return "font-size: 6rem;\n            " + Loader + "{\n                border: 0.2em dotted currentcolor;\n            }";
        default:
            return "font-size: 1rem;\n        " + Loader + "{\n            border: 0.15em dotted currentcolor;\n        }";
    }
};
export var LoaderWrapper = styled.div(templateObject_2 || (templateObject_2 = __makeTemplateObject(["\n    ", "\n    ", "\n    display: ", ";\n"], ["\n    ", "\n    ", "\n    display: ", ";\n"])), function (pr) { return pr.disabled === true ? 'pointer-events: none;opacity: 0.4;' : ''; }, function (pr) { return mixin(pr.size); }, function (pr) { return pr.display ? 'inline-block' : 'none'; });
var templateObject_1, templateObject_2;
