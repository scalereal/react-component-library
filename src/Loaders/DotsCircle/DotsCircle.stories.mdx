import DotsCircle, { SizeMap } from '.';
import { Preview, Story, Meta, Props, ArgsTable, Canvas } from '@storybook/addon-docs/blocks';
import { withKnobs, color, boolean, select } from '@storybook/addon-knobs';

<Meta
    title="Components/Loaders/DotsCircle"
    component={DotsCircle}
    argTypes={{
        size: {
            control: {
                type: 'select',
                options: Object.values(SizeMap)
            }
        },
        Color: {
            control: 'color'
        },
        display: {
            control: {
                type: 'boolean'
            }
        },
        ref: {
            table: {
                disable: true
            }
        },
        theme: {
            table: {
                disable: true
            }
        },
        as: {
            table: {
                disable: true
            }
        }
    }}
/>

# Loaders

Displays a Loader.

export const Template = args => (
    <div>
        <DotsCircle {...args} />
    </div>
);

<Preview>
    <Story name="DotsCircle" args={{ size: 'S' }}>
        {Template.bind({})}
    </Story>
</Preview>

### Color Variations

You can either change the theme or pass the colors as props.

<Preview withToolbar>
    <div style={{ display: 'flex', justifyContent: 'space-around' }}>
        <DotsCircle size="S" Color="green" />
        <DotsCircle size="S" Color="red" />
    </div>
</Preview>

### Sizes

There are Six sizs available `XXS` | `XS` | `S` | `M` | `L` | `XL`.

<Preview withToolbar>
    <div style={{ display: 'flex', justifyContent: 'space-between' }}>
        <DotsCircle size="XXS" />
        <DotsCircle size="XS" />
        <DotsCircle size="S" />
        <DotsCircle size="M" />
        <DotsCircle size="L" />
        <DotsCircle size="XL" />
    </div>
</Preview>

### Props

<Props story="DotsCircle" />
