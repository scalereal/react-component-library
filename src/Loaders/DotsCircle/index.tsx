import * as React from 'react';
import { WithStyle } from '../../utils';
import { Props } from '../types';
import { Loader, LoaderWrapper } from './styled';

export const SizeMap: Props['size'][] = ['XXS','XS', 'S', 'M', 'L', 'XL'];

const DotsCircle:React.FC<Props> & WithStyle = React.memo(React.forwardRef( (props, ref) => {
    const {size} = props;

    return (
        <LoaderWrapper ref={ref} size={size} {...props}>
            <Loader {...props}/>
        </LoaderWrapper>
    );
    
}));

DotsCircle.displayName="DotsCircle";
DotsCircle.defaultProps = {
    size:'S',
    visible: true
};

export default DotsCircle;
