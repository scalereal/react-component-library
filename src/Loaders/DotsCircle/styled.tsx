import styled from 'styled-components';
import { Props } from '../types';

export const Loader = styled.div<Props>`
    display: inline-block;
    width: 1em;
    height: 1em;
    color: ${pr=>pr.Color?pr.Color:'inherit'};
    vertical-align: middle;
    pointer-events: none;
    border-radius: 50%;
    animation: 1s loader-01 linear infinite;

    @keyframes loader-01 {
        0% {
            transform: rotate(0deg);
        }
        100% {
            transform: rotate(360deg);
        }
    }
    
`;

const mixin = (size:any) => {
    switch (size) {
        case "XXS":
            return `font-size: 0.8rem;
            ${Loader}{
                border: 0.188em dotted currentcolor;
            }`;
        case "XS":
            return `font-size: 1rem;
            ${Loader}{
                border: 0.188em dotted currentcolor;
            }
            `;
        case "S":
            return `font-size: 2rem;
            ${Loader}{
                border: 0.2em dotted currentcolor;
            }`;
        
        case "M":
            return `font-size: 3rem;
            ${Loader}{
                border: 0.2em dotted currentcolor;
            }`;

        case "L":
            return `font-size: 5rem;
            ${Loader}{
                border: 0.2em dotted currentcolor;
            }`;
        
        case "XL":
            return `font-size: 6rem;
            ${Loader}{
                border: 0.2em dotted currentcolor;
            }`;
  
      default:
        return `font-size: 1rem;
        ${Loader}{
            border: 0.15em dotted currentcolor;
        }`;
     }
 };

export const LoaderWrapper = styled.div<Props>`
    ${pr => pr.disabled === true?'pointer-events: none;opacity: 0.4;':''}
    ${pr=>  mixin(pr.size)}
    display: ${pr=>pr.visible?'inline-block':'none'};
`;
