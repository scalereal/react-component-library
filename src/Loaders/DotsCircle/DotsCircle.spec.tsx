import { render } from '@testing-library/react';
import * as React from 'react';
import DotsCircle from '.';

describe('DotsCircle Loader', () => {
    it('should render with default theme', () => {
        const { container } = render(<DotsCircle/>);
        expect(container).toMatchSnapshot();
    });
    it('should render with all the props given', () => {
        const { container } = render(
            <DotsCircle
                size="M"
                Color='#126AFA'
            />
        );
        expect(container).toMatchSnapshot();
    });

    it('should render image avatar properly', () => {
        const { container } = render(
            <DotsCircle size="L" />
        );
        expect(container).toMatchSnapshot();
    });

});