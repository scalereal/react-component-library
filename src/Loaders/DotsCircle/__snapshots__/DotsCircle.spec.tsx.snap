// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DotsCircle Loader should render image avatar properly 1`] = `
<div>
  <div
    class="sc-gsTCUz bMFuWA"
  >
    <div
      class="sc-bdfBwQ hSFrdG"
    />
  </div>
</div>
`;

exports[`DotsCircle Loader should render with all the props given 1`] = `
<div>
  <div
    class="sc-gsTCUz lgfpuC"
  >
    <div
      class="sc-bdfBwQ jNUwAr"
    />
  </div>
</div>
`;

exports[`DotsCircle Loader should render with default theme 1`] = `
<div>
  <div
    class="sc-gsTCUz dhEPbz"
  >
    <div
      class="sc-bdfBwQ hSFrdG"
    />
  </div>
</div>
`;
